### ISSUE TITLE
Allow for a customizable "workspace" for the linchpin CLI

### ISSUE THREAD

**Issue by robled** [CentOS-PaaS-SIG/linch-pin#119]
As of right now my **$PWD** before running the linchpin CLI must contain the linchpin topologies/layouts/etc:

```
├── inventory
│   └── cinch_test_topology.inventory
├── keystore
├── layouts
│   ├── my_layout.yml
│   └── openshift-3node-cluster.yml
├── linchpin_config.yml
├── myargs.yml
├── outputs
│   └── cinch_test_topology.output.yaml
├── PinFile
└── topologies
    ├── duffy-3node-cluster.yml
    └── mytopology.yml
```

This "workspace" created by **linchpin init** should be configurable so that the linchpin CLI can be called without needing to be concerned about the **$PWD**.

**Comment by herlo** [comment]
Having the topologies/layouts in the same path is intentional. Some sort of file or environment variable needs to be defined to know where to look. The plan was that linchpin would define sane standard locations, first looking for a PinFile in the current directory of execution. This follows how Vagrant provides a similar Vagrantfile in its location. However, that can be overridden by an environment variable based upon the desire of the user. 

In another issue, we've discussed having an upstream where one could store the topologies/layouts which could then be pulled into the current directory.  Putting this into a workspace should not be a huge concern, but sane defaults should be considered. Thoughts?

**Comment by herlo** [comment]
To the above conversation, here are two issues previously addressing similar parts of this design.

Issue #92 and #96 should be reviewed in reference to this issue.

**Comment by robled** [comment]
I like the "upstream" idea for sure.  But there will be cases where CI users will have some other method to check out the linch-pin workspace, and in that case we will need a way to point linch-pin to the checkout directory without having to "cd" first.  I think the environment variable idea is OK, and that would suit our needs.

**Comment by herlo** [comment]
@robled upstream can be local, so that would likely be the default. I think the $WORKSPACE can be a nice override.

**Comment by herlo** [comment]
Closed by #206

**Comment by herlo** [comment]
![](https://github.trello.services/images/mini-trello-icon.png) [Customizable workspace](https://trello.com/c/y63MNLRB/7-customizable-workspace)
